### ISSUE TITLE
Kodak dataset 在低码率上就达到很高的psnr

### ISSUE THREAD

**Issue by xuezhongcailian** [mx54039q/entroformer#7]
您好，谢谢你的工作和开源。我用您的代码训练sh scripts/train.sh 0.1800，而改变设置depth=2,heads=4，--na unidirectional，--attn_topk -1，并且去掉下面代码
        #if(hasattr(self, 'sos_pred_token')):
        #    sos_pred_token = repeat(self.sos_pred_token, '() n d -> b n d', b = batch_size)
        #    out = torch.cat((sos_pred_token, out[:,:-1,:]), dim=1)
时，训练集为DIV2K 800张图片，随机裁剪训练，我发现改变--alpha(即lamda)的时候，即使变得很大，(如: --alpha 0.18, bpp:0.1684, PSNR:39.04)在Kodak dataset，我的码率很低，psnr很高，请问这是怎么回事，希望得到您的回复，谢谢

**Comment by mx54039q** [comment]
你好, 'sos_pred_token'这个设置是和autoregressive相关的, 用来对第一个像素进行预测的, 然后将token往后移一位. 在unidirectional下这个设置是必需的, 不然会导致context model信息的泄露, 导致'当前pixel预测时, 能够看到他本身'. 在这种设置下, 你的模型是无法正常压缩和解压的. 这应该是造成你训练的模型性能变得很好的原因.

**Comment by xuezhongcailian** [comment]
谢谢。 另有一个疑惑，在代码中计算注意力时，我看到  mask = torch.tril(torch.ones((n, n))).bool()

为什么不是mask = torch.tril(torch.ones((n, n)),  diagonal=-1).bool()

**Comment by mx54039q** [comment]
我已收到你的邮件。                                               by    钱一琛

**Comment by mx54039q** [comment]
> torch.tril(torch.ones((n, n)), diagonal=-1)

如我上一个回复中所述, 已经用sos_pred_token来预测第一个像素, 所以token已经后移一位了, 所以mask的对角线上可以都为1.
